docs/excel/call-excel-apis-from-custom-function.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Call Excel APIs from a custom function
+description: Learn which Excel APIs you can call from your custom function. 
+ms.date: 02/06/2020
+localization_priority: Normal
+---
+
+# Call Excel APIs from a custom function
+
+[!include[Running custom functions in browser runtime note](../includes/excel-shared-runtime-preview-note.md)]
+
+Call Office.js Excel APIs from your custom functions to get range data and obtain more context for your calculations.
+
+Calling Office.js APIs through a custom function can be helpful when:
+
+- A custom function needs to get information from Excel before calculation. This information might include document properties, range formats, custom XML parts, a workbook name, or other Excel-specific information.
+- A custom function will set the cell's number format for the return values after calculation.
+
+[!include[Excel shared runtime note](../includes/note-requires-shared-runtime.md)]
+
+## Code sample
+
+To call into the Office.js APIs you first need a context. Use the `Excel.RequestContext` object to get a context. Then use the context to call the APIs you need in the workbook.
+
+The following code sample shows how to get a range of values from the workbook.
+
+```JavaScript
+/**
+ * @customfunction
+ * @param address range's address
+ **/
+async function getRangeValue (address) {
+ var context = new Excel.RequestContext();
+ var range = context.workbook.worksheets.getActiveWorksheet().getRange(address);
+ range.load();
+ await context.sync();
+ return range.values[0][0];
+}
+```
+
+## Restrictions to calling Office.js through a custom function
```

Comment:
Maybe this section should be called "Best practices when ..." instead of "Restrictions to ...". It sounds like nothing prevents the dev from writing to the workbook, it's just that it's a bad idea to do so.